app/models/zone.rb

```diff
@@ -161,10 +161,9 @@ def availability_zones
   end
 
   def vms_without_availability_zone
-    MiqPreloader.preload(self, :ext_management_systems => :vms)
-    ext_management_systems.flat_map do |e|
-      e.kind_of?(EmsCloud) ? e.vms.select { |vm| vm.availability_zone.nil? } : []
-    end
+    clouds = ext_management_systems.select { |e| e.kind_of?(EmsCloud) }
```

Comment:
I think there's an `ems_clouds` association that you can use here instead of `ext_management_systems`, however that might actually *add* more overhead...not sure.